manifold_airflow_dags/manifold_database_dump_dag.py

```diff
@@ -0,0 +1,103 @@
+"""Airflow DAG to Run the Blogs Sync rake task on a Manifold instance"""
+from datetime import datetime, timedelta
+import airflow
+from airflow.operators.bash_operator import BashOperator
+from airflow.hooks.base_hook import BaseHook
+from airflow.hooks.S3_hook import S3Hook
+from airflow.models import Variable
+from airflow.contrib.operators.ssh_operator import SSHOperator
+from airflow.contrib.operators.sftp_operator import SFTPOperator
+from airflow.contrib.operators.sftp_to_s3_operator import SFTPToS3Operator
+from airflow.operators.python_operator import PythonOperator
+from manifold_airflow_dags.task_slack_posts import slackpostonfail
+from manifold_airflow_dags.task_slack_posts import slackpostonsuccess
+
+MANIFOLD_INSTANCE_SSH_CONN = \
+        BaseHook.get_connection("AIRFLOW_CONN_MANIFOLD_SSH_INSTANCE")
+MANIFOLD_DUMP_DATABASE_INTERVAL = \
+        Variable.get("MANIFOLD_DUMP_DATABASE_SCHEDULE_INTERVAL")
+AIRFLOW_S3 = BaseHook.get_connection("AIRFLOW_S3")
+AIRFLOW_S3_PATH = "manifold"
+AIRFLOW_DATA_BUCKET = \
+        Variable.get("AIRFLOW_DATA_BUCKET")
+
+DUMP_PATH = "/tmp"
+UPLOAD_FILENAME = f"manifold-{ datetime.now().strftime('%Y-%m-%dT%H%z')}.sqlc"
+DEFAULT_ARGS = {
+    'owner': 'airflow',
+    'start_date': datetime(2019, 5, 28),
+    'email': ['steven.ng@temple.edu'],
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 0,
+    'on_failure_callback': slackpostonfail,
+    'retry_delay': timedelta(minutes=5),
+}
+
+DAG = airflow.DAG(
+    'manifold_database_dump',
+    default_args=DEFAULT_ARGS,
+    catchup=False,
+    max_active_runs=1,
+    schedule_interval=MANIFOLD_DUMP_DATABASE_INTERVAL,
+)
+
+#
+# CREATE TASKS
+#
+# Tasks with all logic contained in a single operator can be declared here.
+# Tasks with custom logic are relegated to individual Python files.
+
+SET_DUMP_NAME = PythonOperator(
+    task_id="set_dump_name",
+    python_callable=datetime.now().strftime,
+    op_args=["%Y%m%d_%H%M%S"],
+    dag=DAG
+)
+
+# Note: single quotes are required in the bash command line.
+dump_database_bash = f"""
+sudo su - postgres bash -c \
+  'pg_dump --format=c --file="{ DUMP_PATH }/dbdump_%s.sqlc" manifold'
+""" % "{{ ti.xcom_pull(task_ids='set_dump_name') }}"
+
+DUMP_DATABASE = SSHOperator(
+    task_id='dump_database',
+    command=dump_database_bash,
+    ssh_conn_id="manifold-db",
+    dag=DAG
+)
+
+#
+# Copy the dump file to the worker so we have a local copy
+#
+
+SFTP_GET_DB_DUMP = SFTPOperator(
```

Comment:
Why are you copying the file locally if you are then using SFTPtos3 to transfer the dump to the bucket? We're just going to accumulate db dumps on the worker until we run out of space?

Reply:
Removed